### ISSUE TITLE
Show a prompt when files modified externally on disk (apply, ignore, diff)

### ISSUE THREAD

**Issue by BillDenton** [Microsoft/vscode#14298]
- VSCode Version: 1.6.1
- OS Version: Windows 7 Professional.

Steps to Reproduce:
1. Load a file into VSCode.
2. Edit the same file in another editor.
3. Look at VSCode and see that file has been updated.

Files modified externally on disk are automatically reloaded without warning. This means that I'm not aware that they have changed. It would good to have an indication that this happened with an choice to compare & merge the new and old files, keep the old version or load the new version. I would guess that this would have to be configurable by file or file type (with a global default). For example, SlickEdit shows a prompt to allow reload, cancel (i.e. keep the file in the editor), close or diff.

**Comment by cori** [comment]
As a clarification, _simply_ showing a warning (and **then** overwriting the file?) doesn't really capture the spirit of the problem here, so the resolution needs to do more then just show a warning - it also needs to keep control over what subsequently occurs in the hands of the user, not the application, by virtue of a prompt or something.

**Comment by BillDenton** [comment]
My original comment explains that there a user choice is required so the user can chose "reload, cancel (i.e. keep the file in the editor), close or diff". Also my original title was "Files modified externally on disk are automatically reloaded without warning".

**Comment by v3ss0n** [comment]
Related problem #19826 
The Undo History is also gone if file is changed on disk by another app or extensions.

**Comment by Raydir** [comment]
I'd call this issue very important!

**Problem:**

1. working on a file in VSC for several hours.
2. leave your workplace for lets say going to the toilet for 5mins
3. in the meantime someone is replacing the file (overwrite)
4. come back and go further with your work

--> you wont even notice, that the file has been reloaded if your not really looking at it.

**Comment by siran** [comment]
This issue has more than 1.5years open. Is the team not interested/willing to fix this? I simply can't start using VSCode until this is addressed.

**Comment by euginio** [comment]
Problem persist in last VSCode Version in WS10 Pro. We all lost work every week

**Comment by bpasero** [comment]
I am merging this https://github.com/Microsoft/vscode/issues/23107 which talks about having a setting to control if a change outside of VSCode should bring up a confirmation or not.

@euginio can you please explain how this behaviour means you lost work?

**Others:** would like to understand why it is so important to be notified when a file changed externally? What is the benefit of getting a confirmation dialog popup (probably even modal?) each time a file has changed?

**Comment by siran** [comment]
Is this closed as in "won't do"?

**Comment by bpasero** [comment]
@siran how is my comment in https://github.com/Microsoft/vscode/issues/14298#issuecomment-423176393 indicating that?

**Comment by siran** [comment]
I saw the ticket "closed" and I misuderstood. My comment was an honest question, no subtext. 
![image](https://user-images.githubusercontent.com/180129/45877242-2681ef00-bd6b-11e8-8ee9-8d57a4ba023e.png)

Answering your question: In my experience: being notified if a file changes gives me a greater sense of control/security over the files I have open, when switching branches, for example. Or when a file has been modified by another process or editor. Maybe I'm just watching a log file, and I don't want to change what Im seeing without accepting it, I find deeply disturbing that a file I have open in my editor can change silently without notice.

**Comment by bpasero** [comment]
@siran closed as duplicate of https://github.com/Microsoft/vscode/issues/23107. So this does neither mean it will not happen nor it did happen.

Ok I understand that reasoning, but I am still waiting for someone to explain how this can cause "work to get lost".

**Comment by siran** [comment]
As far as I know, modified files are not updated/changed silently by VS Code, so there is no 'active loss of work' since you had not modified the file. However it could certainly make you waste time, or make to resort to stupid tricks like modifying a file with a space not to have it changed silently, unadvertently and unwillingly by VSCode.
